samza-core/src/main/java/org/apache/samza/storage/blobstore/BlobStoreRestoreManager.java

```diff
@@ -212,9 +213,17 @@ static CompletableFuture<Void> restoreStores(String jobName, String jobId, TaskN
       File loggedBaseDir, StorageConfig storageConfig, BlobStoreRestoreManagerMetrics metrics,
       StorageManagerUtil storageManagerUtil, BlobStoreUtil blobStoreUtil, DirDiffUtil dirDiffUtil,
       ExecutorService executor) {
+    return restoreStores(jobName, jobId, taskName, storesToRestore, prevStoreSnapshotIndexes, loggedBaseDir, storageConfig,
+        metrics, storageManagerUtil, blobStoreUtil, dirDiffUtil, executor, false);
+  }
+
+  public static CompletableFuture<Void> restoreStores(String jobName, String jobId, TaskName taskName, Set<String> storesToRestore,
```

Comment:
Minor: Why public? Make package private and @VisibleForTesting?

Reply:
This is because it's called from CSMUtil directly in case the restore fails. This is separate from restore() which is impl of interface TaskRestoreManager. Both versions of restores are required.